### ISSUE TITLE
nested modules/classes syntax cop

### ISSUE THREAD

**Issue by geniou** [rubocop/rubocop#868]
Basically there are two (or more) ways to nest modules/classes in ruby:

``` ruby
module  Foo
  module Bar
  end
end
```

or

``` ruby
module Foo::Bar
end
```

What do you think a about the idea of having a cop to check the syntax of nested modules/classes. Personally I would prefer the first one. I also heard, that it has some benefits and the ruby-style-guide only use the first one.

Maybe I'm able to write such a cop, but first I wanted to know what you think about this.

**Comment by justin808** [comment]
I put links to many of the articles. I'm still not convinced that compact is incorrect so long as one properly namespaces constants, and classes, rather than depending on the default lookup rules, which i think could get ambiguous.

**Comment by jonas054** [comment]
@abdelbk While it might seem like a good idea that every cop should take care of any violations it might introduce during autocorrect, we can't really do it that way. It would add too much complexity in a lot of cops.

We must rely on other cops to clean up after one cop has autocorrected something. With default configuration, `rubocop -A` corrects the above code into
```ruby
# frozen_string_literal: true

module Module1
  module Module2
    class MyClass
      def my_method; end
    end
  end
end
```
This involves changes made by `Layout/IndentationWidth` among others.

**Comment by abdelbk** [comment]
Thank you @jonas054 for your response.
When I want to apply rubocop on an existing project, I like to activate cops one by one. Therefore, I prefer to see the changes applied by that single cop. Relying on another cop to clean up may introduce changes outside of the scope of the activated cop. It's not a big deal in the grand scheme of things in this case. So, I got my answer.
